### ISSUE TITLE
Lots of warnings when drawing multiple lines

### ISSUE THREAD

**Issue by igamenovoer** [enthought/mayavi#527]
I am using python 3.5 and mayavi 4.5.0 + vtk 7.1 (mayavi‑4.5.0+vtk71‑cp35‑cp35m‑win_amd64.whl downloaded from http://www.lfd.uci.edu/~gohlke/pythonlibs/#mayavi), and I ran into a problem.

I tried to run the draw-many-lines example given by the official documentation, but gets a lot of warnings in the vtkOutputWindow. The lines will be drawn eventually, but it is very slow because you have to wait until all warnings are printed. 

How to fix this? Thanks.

The code is here:
http://docs.enthought.com/mayavi/mayavi/auto/example_plotting_many_lines.html

And the warnings looks like this:
....
Warning: In D:\Build\VTK\VTK-7.1.1\Common\DataModel\vtkPolyData.cxx, line 1023
vtkPolyData (00000150D4F16FE0): Building VTK_TRIANGLE 482 with less than three points, but VTK_TRIANGLE needs at least three points. Check the input.

Warning: In D:\Build\VTK\VTK-7.1.1\Common\DataModel\vtkPolyData.cxx, line 1023
vtkPolyData (00000150D4F16FE0): Building VTK_TRIANGLE 483 with less than three points, but VTK_TRIANGLE needs at least three points. Check the input.

Warning: In D:\Build\VTK\VTK-7.1.1\Common\DataModel\vtkPolyData.cxx, line 1023
vtkPolyData (00000150D4F16FE0): Building VTK_TRIANGLE 484 with less than three points, but VTK_TRIANGLE needs at least three points. Check the input.

Warning: In D:\Build\VTK\VTK-7.1.1\Common\DataModel\vtkPolyData.cxx, line 1023
vtkPolyData (00000150D4F16FE0): Building VTK_TRIANGLE 485 with less than three points, but VTK_TRIANGLE needs at least three points. Check the input.

Warning: In D:\Build\VTK\VTK-7.1.1\Common\DataModel\vtkPolyData.cxx, line 1023
vtkPolyData (00000150D4F16FE0): Building VTK_TRIANGLE 486 with less than three points, but VTK_TRIANGLE needs at least three points. Check the input.
....

**Comment by ying17zi** [comment]
I have the same problem "vtkPolyData (0x7fac677c5290): Building VTK_TRIANGLE 99 with less than three points, but VTK_TRIANGLE needs at least three points. Check the input."

**Comment by prabhuramachandran** [comment]
Thanks for the issue, I can reproduce it and it looks like something changed in VTK 7.1 that breaks this and I'll try to work on a fix next week.

**Comment by ying17zi** [comment]
There is no warning when I create `lines` as `tube` : `lines= mlab.pipeline.tube(src, tube_radius=0.005, tube_sides=6)`.

**Comment by dorisxinyi** [comment]
@ying17zi It works! You saved the day, thanks !!

**Comment by brey** [comment]
Any update on this one? The tube doesn't really work for me.

**Comment by wwols** [comment]
Just add:
`connections = connections.astype(int)`
after
`connections = np.vstack(connections)`

src.mlab_source.dataset.lines expects _int_ array, but in the original code _float_ was created

**Comment by ying17zi** [comment]
> Any update on this one? The tube doesn't really work for me.

Yes. There are warnings again in viewing as `lines`, but no warning as tubes in my example: `$ show -ft f.mat`
These are the links of `show` and `f.mat` for testing:
- [show](https://github.com/ying17zi/show/blob/master/show)
- [f.mat](https://github.com/ying17zi/show/blob/master/data/f.mat)

**Comment by brey** [comment]
I looked into this. Apparently the issue is the stripper filter which is only needed if you want to use tubes. For lines you don't need it. So you can drop the offensive part if you plot only lines. Thus

```python
# The stripper filter cleans up connected lines
#lines = mlab.pipeline.stripper(src)

# Finally, display the set of lines
mlab.pipeline.surface(src, colormap='Accent', line_width=1, opacity=.4)
```

and it works without warnings. 

Cheers.
